actor/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+//! Actor provides a lightweight actor framework based on tokio.
+//!
+//! Features:
+//!  - In memory message passing
+//!  - Strongly typed
+//!  - tracing spans are preserved from caller to actor
+//!
+//! # Example
+//! ```
+#![doc = include_str!("../examples/game/main.rs")]
+#![doc = "```"]
+#![warn(missing_docs)]
+
+mod macros;
+pub use ceramic_actor_macros::*;
+
+pub use tracing;
+
+use snafu::prelude::*;
+use tokio::sync::{mpsc, oneshot};
+
+use async_trait::async_trait;
+use tracing::{debug_span, Span};
+
+/// A message that can be sent to an actor.
+pub trait Message: 'static + Send + Sized {
+    /// The type of results from this message.
+    type Result: 'static + Send;
+
+    /// Returns the messages' type name string
+    fn type_name() -> &'static str {
+        std::any::type_name::<Self>()
+    }
+}
+
+/// Handle a type of [`Message`] for an [`Actor`].
+#[async_trait]
```

Comment:
I hoped we might be able to remove `async_trait` from this crate now that we have a higher MSRV but it's a bit annoying so probably not worth it right now.

Reply:
Yeah, I get a warning that async in public traits is not recommended yet. So I used async_trait.